Solution

DemonWayne opened the pull request.

DemonWayne committed:
Solution

de-vasta commented on src/formatDate.js:
Great job on that! 💪🏼 The solution is neat, concise, and it's easy to understand enough. Got nothing to add (for real) but maybe just some small suggestions to make your code even more flawless...

DemonWayne replied:
Thanks! I've made the requested changes there: https://github.com/mate-academy/js_format_date_advanced/pull/2522/commits/a16912af73b547adc2131059dd9d8c2b37872160

de-vasta commented on src/formatDate.js:
```diff
@@ -8,7 +8,38 @@
+  const dateParts = date.split(fromSeparator);
+
+  fromFormat.slice(0, 3).forEach((part, index) => {
+    if (part === 'YYYY' || part === 'YY') {
```
I'd extract some hardcoded pieces to **constant** variables especially when they're reused more than once (e.g. `'DD'`, `'MM'` etc.)

de-vasta commented on src/formatDate.js:
```diff
@@ -8,7 +8,38 @@
+  const toSeparator = toFormat[toFormat.length - 1];
+
+  const formattedDateParts = toFormat.slice(0, 3).map((part) => {
+    if (part === 'YYYY') {
```
I'd consider to use `switch` (with its tidy structure) when there are solely _strict equality_ here. Or alternatively go with like:
```js
if (<condition>) {return <value>;}
if (<condition>) {return <value>;}
....
return <value>;
```
`else` is a little redundant here anyway, imo

DemonWayne committed:
fix: extract literals & use switch-case

fix: extract overused literals to const's & change logic to switch-case from if-elseif

de-vasta reviewed (approved):
Ok. Fine by me 👍🏽
